HorizonEditor/src/EditorLayer.cpp

```diff
@@ -0,0 +1,194 @@
+#include <pch.h>
+#include "EditorLayer.h"
+
+EditorLayer::EditorLayer(const char* name) :
+	Hzn::Layer(name),
+	m_AspectRatio((float)Hzn::App::getApp().getAppWindow().getWidth() / (float)Hzn::App::getApp().getAppWindow().getHeight()),
+	m_CameraController(Hzn::OrthographicCameraController(m_AspectRatio, 1.0f))
+{
+}
+
+void EditorLayer::onAttach()
+{
+	HZN_TRACE("Editor Layer Attached!");
+	m_CheckerboardTexture = Hzn::Texture2D::create("assets/textures/bear.png");
+	Hzn::FrameBufferProps props;
+	props.width = Hzn::App::getApp().getAppWindow().getWidth();
+	props.height = Hzn::App::getApp().getAppWindow().getHeight();
+
+	m_FrameBuffer = Hzn::FrameBuffer::create(props);
+}
+
+void EditorLayer::onDetach() {}
+
+void EditorLayer::onUpdate(Hzn::TimeStep ts)
+{
+	m_CameraController.onUpdate(ts);
+	m_FrameBuffer->bind();
+	Hzn::RenderCall::setClearColor({ 0.1f, 0.1f, 0.1f, 1.0f });
+	Hzn::RenderCall::submitClear();
+	Hzn::Renderer2D::beginScene((const Hzn::OrthographicCamera&)m_CameraController.getCamera());
+
+	for (int32_t i = 0; i < quads; ++i)
```

Comment:
What are the quads?

Reply:
These quads, it's just a variable that indicates the side of the test grid that I am rendering.

Reply:
The 'quads' reaches scope from EditorLayer.h right?

Reply:
Quads are scope between 5 to 1000, if you see the ImGui slider code, that is what changes this variable.